sdk/keyvault/keyvault-admin/src/generated/keyVaultClient.ts

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is
+ * regenerated.
+ */
+
+import * as coreHttp from "@azure/core-http";
+import * as Models from "./models";
+import * as Mappers from "./models/mappers";
+import * as Parameters from "./models/parameters";
+import * as operations from "./operations";
+import { KeyVaultClientContext } from "./keyVaultClientContext";
+
+class KeyVaultClient extends KeyVaultClientContext {
+  // Operation groups
+  roleDefinitions: operations.RoleDefinitions;
+  roleAssignments: operations.RoleAssignments;
+
+  /**
+   * Initializes a new instance of the KeyVaultClient class.
+   * @param apiVersion Client API version.
+   * @param [options] The parameter options
+   */
+  constructor(apiVersion: string, options?: coreHttp.ServiceClientOptions) {
+    super(apiVersion, options);
+    this.roleDefinitions = new operations.RoleDefinitions(this);
+    this.roleAssignments = new operations.RoleAssignments(this);
+  }
+
+  /**
+   * Creates a full backup using a user-provided SAS token to an Azure blob storage container.
+   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+   * @param [options] The optional parameters
+   * @returns Promise<Models.FullBackupResponse>
+   */
+  fullBackup(vaultBaseUrl: string, options?: Models.KeyVaultClientFullBackupOptionalParams): Promise<Models.FullBackupResponse>;
```

Comment:
All the methods on this client are about backup and restore. 
How does the RBAC support factor in?

Reply:
There's a class named `RoleAssignments` in this PR that I believe deals with this. I'm using pretty much what goes out of swagger, so I'm relying on swagger (?)

Reply:
Looks as expected at a high level to me from a generation perspective. However, even at the convenience layer we created separate service clients for RBAC and Backup for .Net.

Reply:
Ah, found it. One would do `client.roleAssignments.` to find the role related methods